core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -337,16 +344,34 @@ public void setBuddyAcceptedWithSecretUserInfo(UUID buddyID, UUID userAnonymized
 		return buddyIDs.stream().map(id -> getBuddy(id)).collect(Collectors.toSet());
 	}
 
+	private Set<Buddy> getBuddyEntitiesOfUser(UUID forUserID)
+	{
+		UserDTO user = userService.getPrivateUser(forUserID);
+		return getBuddyEntities(user.getPrivateData().getBuddyIDs());
+	}
+
+	private Set<Buddy> getBuddyEntities(Set<UUID> buddyIDs)
+	{
+		return buddyIDs.stream().map(id -> getEntityByID(id)).collect(Collectors.toSet());
+	}
+
 	private void removeMessagesSentByBuddy(User user, Buddy buddy)
 	{
-		Optional<UUID> userAnonymizedID = buddy.getUserAnonymizedID();
-		if (!userAnonymizedID.isPresent())
+		Optional<UUID> buddyUserAnonymizedID = buddy.getUserAnonymizedID();
+		if (!buddyUserAnonymizedID.isPresent())
 		{
 			return;
 		}
-		removeNamedMessagesSentByUser(user, userAnonymizedID.get());
+		removeNamedMessagesSentByUser(user, buddyUserAnonymizedID.get());
 		UserAnonymizedDTO userAnonymized = userAnonymizedService.getUserAnonymized(user.getUserAnonymizedID());
-		removeAnonymousMessagesSentByUser(userAnonymized, userAnonymizedID.get());
+		removeAnonymousMessagesSentByUser(userAnonymized, buddyUserAnonymizedID.get());
+	}
+
+	private void removeMessagesSentByUserToBuddy(User user, Buddy buddy)
+	{
+		UUID buddyUserAnonymizedID = getUserAnonymizedIDForBuddy(buddy);
+		UserAnonymizedDTO buddyUserAnonymized = userAnonymizedService.getUserAnonymized(buddyUserAnonymizedID);
+		removeAnonymousMessagesSentByUser(buddyUserAnonymized, user.getUserAnonymizedID());
```

Comment:
> Why only the anonymous messages? I would expect the named messages too.

Added an explanatory comment:
```
// We are not removing the named messages because we don't have User entity anymore
// (this method is being called from removeBuddyInfoForRemovedUser) and thus we don't know the named destination.
// Given that the user and the named destination are both removed, this is not causing any issues.
```